app/frontend/src/features/messages/groupchats/GroupChatView.tsx

```diff
@@ -1,59 +1,54 @@
-import { Box, BoxProps, IconButton, Typography } from "@material-ui/core";
+import { Box, Typography } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
 import CloseIcon from "@material-ui/icons/Close";
 import * as React from "react";
-import { useCallback, useEffect, useState } from 'react';
+import { useEffect } from "react";
+import {
+  fetchMessagesThunk,
+  leaveGroupChatThunk,
+  sendMessageThunk,
+  setGroupChat,
+} from ".";
 import Alert from "../../../components/Alert";
+import Button from "../../../components/Button";
 import CircularProgress from "../../../components/CircularProgress";
-import { GroupChat, Message } from "../../../pb/conversations_pb";
-import { service } from "../../../service";
+import { useAppDispatch, useTypedSelector } from "../../../store";
 import MessageList from "../messagelist/MessageList";
 
 const useStyles = makeStyles({ root: {} });
 
-export interface GroupChatViewProps extends BoxProps {
-  groupChat: GroupChat.AsObject;
-  handleClose: () => void;
-}
+export default function GroupChatView() {
+  const dispatch = useAppDispatch();
+  const { error, loading, groupChat, messages } = useTypedSelector((state) => {
+    return state.groupChats.groupChatView;
+    // if (componentState.groupChat) {
+    //   return { ...componentState, groupChat: componentState.groupChat };
+    // } else {
+    //   throw new Error("No groupChat");
+    // }
+  });
+
+  const handleSend = (text: string) =>
+    dispatch(sendMessageThunk({ groupChat: groupChat!, text }));
 
-export default function GroupChatView({
-  groupChat,
-  handleClose,
-}: GroupChatViewProps) {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState("");
-  const [messages, setMessages] = useState<Message.AsObject[]>([]);
+  const closeGroupChat = () => dispatch(setGroupChat(null));
 
-  const handleSend = async (text: string) => {
-    await service.conversations.sendMessage(groupChat.groupChatId, text);
-    await fetchMessages();
-  };
+  const leaveGroupChat = () => dispatch(leaveGroupChatThunk(groupChat!));
 
-  const fetchMessages = useCallback(async () => {
-    try {
-      const messages = await service.conversations.getGroupChatMessages(
-        groupChat.groupChatId
-      );
-      setMessages(messages);
-    } catch (error) {
-      setError(error.message);
-    }
-  }, [groupChat.groupChatId]);
+  useEffect(() => void dispatch(fetchMessagesThunk(groupChat!)), []);
```

Comment:
Why not move this to `setGroupChat` or some thunk?

Reply:
What do you mean? We cannot move useEffect in a thunk. We need the useEffect to load messages on opening the view.

Reply:
Ah, you mean loading the messages on setting the groupChat. Yes, that is also possible. Not sure what is best.

Reply:
Now it is removed.